Formula/codestore.rb

```diff
@@ -0,0 +1,18 @@
+class Codestore < Formula
+  desc "Command-line interface of code.store"
+  homepage "https://code.store/"
+  url "https://s3.eu-central-1.amazonaws.com/s3.code.store/codestore-v1.2.12/codestore-v1.2.12.tar.gz"
+  sha256 "7474ac084e12cdd946b30e2eed7e5689cfbc656ff19e53f947141e67fa3c0d28"
+  license "MIT"
+  depends_on "node"
+
+  def install
+    inreplace "bin/codestore", /^CLIENT_HOME=/, "export CODESTORE_OCLIF_CLIENT_HOME=#{lib/"client"}\nCLIENT_HOME="
+    libexec.install Dir["*"]
+    bin.install_symlink libexec/"bin/codestore"
+  end
+
+  test do
+    system bin/"codestore", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.